File: src/main/java/com/zakpruitt/pbst/entities/GradingSubmission.java
package com.zakpruitt.pbst.entities;

import com.zakpruitt.pbst.enums.GradingCompany;
import com.zakpruitt.pbst.enums.SubmissionStatus;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class GradingSubmission {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String submissionName;
    private LocalDate sendDate;
    private LocalDate returnDate;

    @Enumerated(EnumType.STRING)
    private GradingCompany company;

    @Enumerated(EnumType.STRING)
    private SubmissionStatus status;

    private String submissionMethod; // e.g., "GameStop", "Self"

    @Column(columnDefinition = "TEXT")
    private String notes;

    private BigDecimal totalGradingCost;

    @OneToMany(mappedBy = "gradingSubmission")
    private List<TrackedItem> items = new ArrayList<>();
}

File: internal/models/grading_submission.go
package models

import (
	"database/sql"
	"time"

	"gorm.io/gorm"
)

type GradingSubmission struct {
	ID               uint           `gorm:"primaryKey" json:"id"`
	SubmissionName   string         `gorm:"column:submission_name" json:"submission_name"`
	SendDate         sql.NullTime   `gorm:"column:send_date" json:"send_date"`
	ReturnDate       sql.NullTime   `gorm:"column:return_date" json:"return_date"`
	Company          string         `json:"company"`
	Status           string         `json:"status"`
	SubmissionMethod string         `gorm:"column:submission_method" json:"submission_method"`
	Notes            sql.NullString `gorm:"type:text" json:"notes"`
	TotalGradingCost float64        `gorm:"column:total_grading_cost" json:"total_grading_cost"`
	Items            []TrackedItem  `gorm:"foreignKey:GradingSubmissionID" json:"items"`
	CreatedAt        time.Time      `json:"created_at"`
	UpdatedAt        time.Time      `json:"updated_at"`
	DeletedAt        gorm.DeletedAt `gorm:"index" json:"-"`
}
